lib/dbservice_web/controllers/group_user_controller.ex

```diff
@@ -121,4 +122,36 @@ defmodule DbserviceWeb.GroupUserController do
       send_resp(conn, :no_content, "")
     end
   end
+
+  defp create_new_group_user(conn, params) do
+    group = Groups.get_group!(params["group_id"])
+
+    enrollment_record = %{
+      "group_id" => group.child_id,
+      "group_type" => group.type,
+      "user_id" => params["user_id"],
+      "grade_id" => params["grade_id"],
+      "academic_year" => params["academic_year"],
+      "start_date" => params["start_date"]
+    }
+
+    with {:ok, %EnrollmentRecord{} = _} <-
+           EnrollmentRecords.create_enrollment_record(enrollment_record) do
+      with {:ok, %GroupUser{} = group_user} <- GroupUsers.create_group_user(params) do
+        conn
+        |> put_status(:created)
+        |> put_resp_header("location", Routes.group_user_path(conn, :show, group_user))
+        |> render("show.json", group_user: group_user)
+      end
+    end
+  end
+
+  defp update_existing_group_user(conn, existing_group_user, params) do
+    with {:ok, %GroupUser{} = group_user} <-
+           GroupUsers.update_group_user(existing_group_user, params) do
+      conn
+      |> put_status(:ok)
+      |> render("show.json", group_user: group_user)
+    end
+  end
```

Comment:
When updating an existing `group_user`, shouldn't we also update `enrollment_record` as well?

Reply:
Yes, thats why I started that discussion on discord.

Reply:
Not implementing this at the moment, until discussion is resolved.